src/Cli/dotnet/commands/dotnet-workload/install/NetSdkMsiInstallerClient.cs

```diff
@@ -182,6 +185,66 @@ public void GarbageCollect(Func<string, IWorkloadResolver> getResolverForWorkloa
             }
         }
 
+        public string InstallWorkloadSet(string path)
+        {
+            return ModifyWorkloadSet(path, InstallAction.Install);
+        }
+
+        public void RollBackWorkloadSetInstallation()
+        {
+            if (_newWorkloadSetPath is not null)
+            {
+                ModifyWorkloadSet(_newWorkloadSetPath, InstallAction.Uninstall);
+                if (_previousWorkloadSetVersion is not null)
+                {
+                    File.WriteAllText(Path.Combine(_newWorkloadSetPath, "version.txt"), _previousWorkloadSetVersion);
```

Comment:
What is this version.txt file?  Is this writing to the advertising manifest package?

Reply:
Yes—this was building off of a suggestion from @MiYanni that I rather liked. When we're initially downloading the workload set package, we know what version we ended up downloading, but when it comes time to actually use it, that information is gone, and I couldn't find an easy way to get it back, so at that point, I instead create a version.txt file in the advertising manifest package (which is basically a scratch package, as far as I'm concerned) and parse it later to get the version back. Here, I'm taking advantage of that mechanism to pretend that we'd written the previous workload set version into the version.txt file, which means that when it goes to install a new workload set, it will install the previous one—that is, the one specified in that version file.

Reply:
Make it more explicit on kind of version (packageVersion vs. workloadSetVersion)